libopae/plugins/xfpga/enum.c

```diff
@@ -477,134 +478,67 @@ STATIC fpga_result enum_afu(const char *sysfspath, const char *name,
 	return FPGA_OK;
 }
 
-STATIC fpga_result enum_top_dev(const char *sysfspath, struct dev_list *list,
-			 bool include_port)
-{
-	fpga_result result = FPGA_NOT_FOUND;
-	struct stat stats;
 
-	struct dev_list *pdev;
+STATIC fpga_result enum_fpga_region_resources(struct dev_list *list,
+				bool include_port)
+{
+	fpga_result result                 = FPGA_NOT_FOUND;
+	struct dev_list *pdev              = NULL;
+	int i                              = 0;
+	int region_count                   = 0 ;
+	sysfs_fpga_region *region          = NULL;
 
-	DIR *dir;
-	struct dirent *dirent;
-	char spath[SYSFS_PATH_MAX];
-	int res;
-	char *p;
-	int f;
-	unsigned s, b, d;
+	region_count = sysfs_region_count();
 
-	// Make sure it's a directory.
-	if (stat(sysfspath, &stats) != 0) {
-		FPGA_MSG("stat failed: %s", strerror(errno));
+	if (region_count <= 0) {
+		FPGA_MSG("Not found fpga region's");
 		return FPGA_NO_DRIVER;
 	}
 
-	if (!S_ISDIR(stats.st_mode))
-		return FPGA_OK;
-
-	res = readlink(sysfspath, spath, sizeof(spath));
-	if (-1 == res) {
-		FPGA_MSG("Can't read link");
-		return FPGA_NO_DRIVER;
-	}
+	for (i = 0; i < region_count; i++) {
 
-	pdev = add_dev(sysfspath, "", list);
-	if (!pdev) {
-		FPGA_MSG("Failed to allocate device");
-		return FPGA_NO_MEMORY;
-	}
+		region  = (sysfs_fpga_region *)sysfs_get_region(i);
```

Comment:
If declared as const above, there should be no need to cast here.

Reply:
fixed